app/assets/javascripts/discourse/app/components/topic-list/header.gjs

```diff
@@ -1,26 +1,40 @@
-const Header = <template>
-  <tr>
-    {{#each @columns as |entry|}}
-      <entry.value.header
-        @sortable={{@sortable}}
-        @activeOrder={{@order}}
-        @changeSort={{@changeSort}}
-        @ascending={{@ascending}}
-        @category={{@category}}
-        @name={{@listTitle}}
-        @bulkSelectEnabled={{@bulkSelectEnabled}}
-        @showBulkToggle={{@toggleInTitle}}
-        @canBulkSelect={{@canBulkSelect}}
-        @canDoBulkActions={{@canDoBulkActions}}
-        @showTopicsAndRepliesToggle={{@showTopicsAndRepliesToggle}}
-        @newListSubset={{@newListSubset}}
-        @newRepliesCount={{@newRepliesCount}}
-        @newTopicsCount={{@newTopicsCount}}
-        @bulkSelectHelper={{@bulkSelectHelper}}
-        @changeNewListSubset={{@changeNewListSubset}}
-      />
-    {{/each}}
-  </tr>
-</template>;
+import Component from "@glimmer/component";
+import { applyValueTransformer } from "discourse/lib/transformer";
 
-export default Header;
+export default class Header extends Component {
+  get sortable() {
+    return applyValueTransformer(
+      "topic-list-header-sortable-column",
+      this.args.sortable,
+      {
+        category: this.args.category,
+        name: this.args.name,
+      }
+    );
+  }
```

Comment:
Technically I think we could keep this as a template-only component, using syntax like this?

```
{{applyValueTransformer @sortable (hash @category @name)}}
```